cmd/cmd_perform_device_flow.go

```diff
@@ -0,0 +1,108 @@
+// Copyright © 2022 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ory/hydra/v2/cmd/cliclient"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/x/cmdx"
+	"github.com/ory/x/flagx"
+	"github.com/ory/x/urlx"
+)
+
+func NewPerformDeviceCodeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "device-code",
+		Example: "{{ .CommandPath }} --client-id ... --client-secret ...",
+		Short:   "An exemplary OAuth 2.0 Client performing the OAuth 2.0 Device Code Flow",
+		Long: `Performs the device code flow. Useful for getting an access token and an ID token in machines without a browser.
+		The client that will be used MUST support the "client_secret_post" token-endpoint-auth-method
```

Comment:
Perhaps we should document that the client must support this method to a better place, where it will be more clear to the user.